src/command-line/uninstall.js

```diff
@@ -28,45 +28,75 @@ program
 
 		if (!fs.existsSync(packagesConfig)) {
 			log.warn(packageWasNotInstalled);
-			return;
+			process.exit(1);
 		}
 
-		const npm = child.spawn(
-			process.platform === "win32" ? "npm.cmd" : "npm",
+		const npm = process.platform === "win32" ? "npm.cmd" : "npm";
+		const errorHandler = (error) => {
+			log.error(
+				`Failed to uninstall ${colors.green(packageName)}. ` +
+				`${typeof x === "number" ? "Exit code" : "Error"}: ${error}`
+			);
+			process.exit(1);
+		};
+
+		// First, we check if the package is installed with `npm list`
+		const list = child.spawn(
+			npm,
 			[
-				"uninstall",
-				"--no-progress",
+				"list",
+				"--depth",
+				"0",
 				"--prefix",
 				packagesParent,
 				packageName,
 			],
 			{
-				// This is the same as `"inherit"` except `process.stdout` is piped
-				stdio: [process.stdin, "pipe", process.stderr],
+				// This is the same as `"inherit"` except:
+				// - `process.stdout` is piped so we can test if the output mentions the
+				//   package was found
+				// - `process.stderr` is ignored to silence `npm ERR! extraneous` errors
+				stdio: [process.stdin, "pipe", "ignore"],
 			}
 		);
 
-		let hasUninstalled = false;
-
-		npm.stdout.on("data", () => {
-			hasUninstalled = true;
```

Comment:
This is the key change. No more assumption that if `uninstall` outputted something, then package was uninstalled, because that is not true on npm v5 (shipping with Node v8) anymore.

Instead, we're checking once and for all with `npm list` before uninstalling. I should have done that in the first place, but I was lazy and thought we could get smart about it. It's npm, so we can't lol.